### PR TITLE
Additional Input and Math Nodes

### PR DESCRIPTION
**Pull request by Ezbaze**
image

### CODE REVIEW

File: node-graph/interpreted-executor/src/node_registry.rs [hunk 1]
```diff
@@ -186,6 +186,14 @@ fn node_registry() -> HashMap<NodeIdentifier, HashMap<NodeIOTypes, NodeConstruct
...
 		register_node!(graphene_core::ops::AddNode, input: (u32, u32), params: []),
 		register_node!(graphene_core::ops::AddNode, input: (u32, &u32), params: []),
+		register_node!(graphene_core::ops::SubtractNode, input: (u32, u32), params: []),
+		register_node!(graphene_core::ops::SubtractNode, input: (u32, &u32), params: []),
+		register_node!(graphene_core::ops::DivideNode, input: (u32, u32), params: []),
+		register_node!(graphene_core::ops::DivideNode, input: (u32, &u32), params: []),
+		register_node!(graphene_core::ops::MultiplyNode, input: (u32, u32), params: []),
+		register_node!(graphene_core::ops::MultiplyNode, input: (u32, &u32), params: []),
+		register_node!(graphene_core::ops::ModuloNode, input: (u32, u32), params: []),
+		register_node!(graphene_core::ops::ModuloNode, input: (u32, &u32), params: []),
```
Comment (TrueDoctor) [inline node-graph/interpreted-executor/src/node_registry.rs:196]: We should make a macro for this

File: node-graph/gcore/src/ops.rs [hunk 2]
```diff
@@ -30,18 +31,151 @@ where
...
+	fn eval(&'i self, input: (L, R)) -> Self::Output {
+		input.0.pow(input.1)
+	}
+}
+
+impl ExponentNode {
+	pub const fn new() -> Self {
+		Self
+	}
+}
```
Comment (TrueDoctor) [inline node-graph/gcore/src/ops.rs:136]: I would suggest you don't even bother implementing the non Parameter variants as they are not used anywhere outside of test cases

File: node-graph/gcore/src/ops.rs [hunk 3]
```diff
@@ -30,18 +31,151 @@ where
...
+
+pub struct ExponentParameterNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(ExponentParameterNode)]
+fn flat_map<U, T>(first: U, second: T) -> <U as Pow<T>>::Output
```
Comment (TrueDoctor) [inline node-graph/gcore/src/ops.rs:143]:
```suggestion
fn exp<U, T>(first: U, second: T) -> <U as Pow<T>>::Output
```

**Review by TrueDoctor — CHANGES_REQUESTED** [review]

File: node-graph/gcore/src/ops.rs [hunk 4]
```diff
@@ -30,18 +31,71 @@ where
...
+	first / second
+}
+
+// Multiply
+pub struct MultiplyParameterNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(MultiplyParameterNode)]
 fn flat_map<U, T>(first: U, second: T) -> <U as Mul<T>>::Output
```
Comment (TrueDoctor) [inline node-graph/gcore/src/ops.rs:66]:
```suggestion
fn mul<U, T>(first: U, second: T) -> <U as Mul<T>>::Output
```

File: node-graph/gcore/src/ops.rs [hunk 5]
```diff
@@ -30,18 +31,71 @@ where
...
+	first.pow(second)
+}
+
+// Modulo
+pub struct ModuloParameterNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(ModuloParameterNode)]
+fn flat_map<U, T>(first: U, second: T) -> <U as Rem<T>>::Output
```
Comment (TrueDoctor) [inline node-graph/gcore/src/ops.rs:92]:
```suggestion
fn mod<U, T>(first: U, second: T) -> <U as Rem<T>>::Output
```

Comment (Ezbaze) [reply node-graph/gcore/src/ops.rs]: changed from "mod" to "modulo" as the build was failing

**Review by TrueDoctor — APPROVED** [review]

**Comment by TrueDoctor** [comment]
We can do the macroification of the node registry entries later

**Pull request merged**